src/tasks/mod.rs

```diff
@@ -351,7 +417,7 @@ where
                     .expect("Failed to receive broadcast messages"),
             );
             async_sleep(Duration::new(0, 500)).await;
-            network.shut_down().await;
+            // network.shut_down().await;
```

Comment:
I'm a bit unsure about this. Don't we still need to shut down the two `network`s somewhere else?

Reply:
Yes, we should shut down them down somewhere else, though not sure exactly where